java-checks/src/main/java/org/sonar/java/checks/RandomFloatToIntCheck.java

```diff
@@ -63,19 +63,26 @@ public List<Tree.Kind> nodesToVisit() {
   @Override
   public void visitNode(Tree tree) {
     TypeCastTree castTree = (TypeCastTree) tree;
-    if(castTree.type().symbolType().is("int")) {
-      castTree.expression().accept(new RandomDoubleVisitor());
+    if (castTree.type().symbolType().is("int")) {
+      castTree.expression().accept(new RandomDoubleVisitor("nextInt()"));
+    } else if (castTree.type().symbolType().is("long")) {
+      castTree.expression().accept(new RandomDoubleVisitor("nextLong()"));
```

Comment:
If we are going to check the symbol  1 to 2 times, we might as well extract it as a variable
```suggestion
    SymbolType symbolType = castTree.type().symbolType().is("int");
    if (symbolType.is("int")) {
      castTree.expression().accept(new RandomDoubleVisitor("nextInt()"));
    } else if (symbolType.is("long")) {
      castTree.expression().accept(new RandomDoubleVisitor("nextLong()"));
```